### ISSUE TITLE
Updating Brave causes session to be lost

### ISSUE THREAD

**Issue by vertigo220** [brave/brave-browser#3491]
## Description 

After updating to most recent version (Version 0.60.45 Chromium: 72.0.3626.109 (Official Build) (64-bit)), Brave failed to restart as well as restore previous session. After telling it to restart then waiting a few minutes, I manually started it, and then my tabs from my previous session (before updating/restarting) were gone, despite having it set in options to load them.

## Steps to Reproduce

   1. Go to Settings > On Startup and enable the "Continue where you left off" option
   2. Open several tabs
   3. Update Brave
   4. Restart Brave

## Actual result:

See description

## Expected result:

Brave should actually restart, and the option should be honored (tabs that were open before should be reopened).

## Reproduces how often: 

Not sure, just updated and it happened.

## Brave version (brave://version info)

See above.

### Reproducible on current release:
- Does it reproduce on brave-browser dev/beta builds? 

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by GeetaSarvadnya** [comment]
Upgraded profile from `0.59.35` to `0.60.45` Windows 10 x64 - Unable to reproduce
Upgraded profile from `0.58.21` to `0.60.45` Windows 10 x64 - Unable to reproduce

**Comment by rebron** [comment]
cc: @kjozwiak to check

**Comment by Kinantb** [comment]
This happened to me recently as well

**Comment by GeetaSarvadnya** [comment]
Verified upgrade scenarios in Windows 10 x64. 

1. Upgraded from  `0.64.77` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
2. Upgraded from  `0.63.55` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
3. Upgraded from  `0.63.52` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
4. Upgraded from  `0.63.48` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
5. Upgraded from  `0.63.46` to `0.65.118` - Working as expected, did not lose any sessions after upgrade

**Comment by Kinantb** [comment]
> Verified upgrade scenarios in Windows 10 x64.
> 
> 1. Upgraded from  `0.64.77` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
> 2. Upgraded from  `0.63.55` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
> 3. Upgraded from  `0.63.52` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
> 4. Upgraded from  `0.63.48` to `0.65.118` - Working as expected, did not lose any sessions after upgrade
> 5. Upgraded from  `0.63.46` to `0.65.118` - Working as expected, did not lose any sessions after upgrade

Well this happened on Ubuntu 18.04 (sorry I forgot to mention)

**Comment by vertigo220** [comment]
I don't know what version it was on when it happened. I don't necessarily upgrade every version, especially on my Surface (Win 10, btw), and I don't really pay attention to the version numbers, especially the old one. And just because it doesn't happen in your testing, doesn't mean it can't happen.

**Comment by kjozwiak** [comment]
@Kinantb @vertigo220 when you run into this issue and lose tabs/windows, have you tried restoring them from the `Recently Closed` section under `History`? Sometimes when Brave doesn't re-open all the previous windows/tabs after an update, you should be able to restore them from the `Recently Closed` section.

**Comment by Kinantb** [comment]
@kjozwiak yes I tried to do so but that list only contained the recently opened ones. As the ones who are delayed-loaded on demand (which made the majority of my tabs) are not counted as "recently closed".

**Comment by vertigo220** [comment]
It was too long ago to remember. I rarely use Brave, so it's possible it might be a more frequent issue (for me at least) but I just don't see it because I don't use it much.

**Comment by bsclifton** [comment]
I've had this happen to me- but only when Brave crashes. When that happens, it seems to lose the history from that session (so you can't just re-open recently closed windows, since they don't show)

**Comment by Kinantb** [comment]
This happened to me again recently, but thankfully now I have a session manager to restore my tabs, otherwise I would have just left the browser for good because it is a very annoying problem

**Comment by kjozwiak** [comment]
> but thankfully now I have a session manager to restore my tabs

Just curious, do you mean the Brave session manager or a third party extension?

**Comment by Kinantb** [comment]
> > but thankfully now I have a session manager to restore my tabs
> 
> Just curious, do you mean the Brave session manager or a third party extension?

Third party

**Comment by flyingscot5** [comment]
This issue still happens on Brave version 1.0.1

**Comment by tnettle** [comment]
Still happening in 2020 with Brave version Version 1.3.115 Chromium: 80.0.3987.87 (Official Build) (64-bit)  on Windows 10 . Must be a fix coming soon.

**Comment by ljosephpratt** [comment]
Still happening.

![Screen Shot 2020-06-25 at 5 33 18 PM](https://user-images.githubusercontent.com/32916356/85797650-0cc4c700-b70a-11ea-9951-e7c82dca849a.png)

[Brave Browser_2020-06-25-171615_WEX-W700077.crash.txt](https://github.com/brave/brave-browser/files/4833917/Brave.Browser_2020-06-25-171615_WEX-W700077.crash.txt)

**Comment by bsclifton** [comment]
Last comments from over 5 years ago. Closing as stale. If you're running into this problem, let's please open a new issue. Thanks!
